test/test_losses.py

```diff
@@ -0,0 +1,229 @@
+import pytest
+import torch
+import torch.nn.functional as F
+from common_utils import cpu_and_gpu
+from torchvision import ops
+
+
+def get_boxes(dtype, device):
+    box1 = torch.tensor([-1, -1, 1, 1], dtype=dtype, device=device)
+    box2 = torch.tensor([0, 0, 1, 1], dtype=dtype, device=device)
+    box3 = torch.tensor([0, 1, 1, 2], dtype=dtype, device=device)
+    box4 = torch.tensor([1, 1, 2, 2], dtype=dtype, device=device)
+
+    box1s = torch.stack([box2, box2], dim=0)
+    box2s = torch.stack([box3, box4], dim=0)
+
+    return box1, box2, box3, box4, box1s, box2s
+
+
+def assert_iou_loss(iou_fn, box1, box2, expected_loss, dtype, device, reduction="none"):
+    tol = 1e-3 if dtype is torch.half else 1e-5
```

Comment:
Is this still needed? [`torch.testing.assert_close`](https://pytorch.org/docs/master/testing.html#torch.testing.assert_close) uses the following default tolerances

| `dtype`   | `rtol`   | `atol`   |
|-----------|----------|----------|
| `float16` | `1e-3`   | `1e-5`   |
| `float32` | `1.3e-6` | `1e-5`   |
| `float64` | `1e-7`   | `1e-7`   |

They are a little stricter than what we have right now. Could you try if the tests also pass without specifying the tolerances?

This comment applies to all occurrences of setting the tolerances.